utils/styler.py

```diff
@@ -1,12 +1,193 @@
 import logging
 import os
-from pathlib import Path
+import numpy as np
 
-from qgis.core import QgsMapLayerStyle
+from typing import List
+from pathlib import Path
+from qgis.PyQt.QtGui import QColor
+from qgis.core import (
+    QgsGradientColorRamp,
+    QgsGradientStop,
+    QgsMapLayerStyle,
+    QgsStyle,
+    QgsVectorLayer
+)
+from qgis.utils import iface
+from ThreeDiToolbox.utils.color import (
+    ColorRampData,
+    COLOR_RAMP_OCEAN_DEEP,
+    COLOR_RAMP_OCEAN_HALINE,
+    COLOR_RAMP_OCEAN_CURL
+)
+from ThreeDiToolbox.datasource.result_constants import WET_CROSS_SECTION_AREA
 
 logger = logging.getLogger(__name__)
 
 STYLES_ROOT = Path(__file__).parent.parent / "layer_styles"
+ANIMATION_LAYERS_NR_LEGEND_CLASSES = 24
+assert ANIMATION_LAYERS_NR_LEGEND_CLASSES % 2 == 0
+
+def color_ramp_from_data(data: ColorRampData):
+    color1 = QColor(data.colors[0])
+    color2 = QColor(data.colors[-1])
+    stops = []
+    for i, color in enumerate(data.colors[1:-1]):
+        stop = QgsGradientStop((i + 1) / (len(data.colors) - 1), QColor(color))
+        stops.append(stop)
+
+    ramp = QgsGradientColorRamp(color1=color1, color2=color2, stops=stops)
+    ramp.setInfo(data.info)
+    return ramp
+
+
+def add_color_ramp(data: ColorRampData):
+    """Add color ramp to QGIS or replace if exists"""
+
+    QgsStyle.defaultStyle().addColorRamp(data.name, color_ramp_from_data(data))
+    # If ramp with this name already exists, it will be overridden (default QGIS API behaviour)
+
+
+def style_animation_flowline_current(
+        lyr: QgsVectorLayer,
+        class_bounds: List[float],
+        variable
+):
+    """Applies styling to Animation Toolbar flowline layer in 'current' mode"""
+
+    # Load basic style settings from qml file
+    qml_path = STYLES_ROOT / 'tools' / 'animation_toolbar' / 'flowline_current.qml'
+    lyr.loadNamedStyle(str(qml_path))
+    ren = lyr.renderer()
+
+    # Set correct legend symbol rotation
+    sym = ren.sourceSymbol().clone()
```

Comment:
Why not `renderer` and `symbol` instead of the abbreviation?
(Unless it is normal in qgis code: I'm slowly starting to get used to `lyr`...)

Reply:
I think it's pretty common from the examples I've seen, but I'll change it to fully written